[Ready][?] Robustin's cult nerf, back from the dead.

lilacrhododendron opened the pull request:
:cl: Robustin and Subject217
balance: Once the blood cult reaches 20% of the active player population, they will receive a notice that the cult is "rising" - and after a moderate delay the eyes of all existing and new cultists will be permanently red. Examining a cultist with uncovered eyes will confirm their supernatural appearance.
balance: Once the blood cult reaches 40% of the active player population they will receive a warning and after a moderate delay will glow red, permanently revealing the identity of existing and new blood cultists.
add: Any non-cultist can now strike a (filled) soulshard with a bible to purify it. Purified shades have a unique appearance and will be eager to get revenge on the cult. Bibles can be printed in the library.
balance: Juggernauts have 25% less HP and 35% less projectile reflection.
balance: Cult mirror shield is slightly easier to break, has less illusions, and has -33% throwing distance.
balance: The EMP blood spell has -1 light and heavy radius.
balance: The revive rune now requires 3 sacrifices per revive, it still starts with one "freebie" revive. Giving "souls" to AFK/Catatonic cultists remains free of charge.
balance: Twisted Construction now has a channel time with added noise/effects when used on doors. It also hurts you slightly more to use.
balance: You can now only hold 1 blood spell without an empowering rune and 4 with one.
del: The Bloody Bastard Sword is no longer available in Blood Cult. It's still spawnable with admin tools.
/:cl:

I really liked these changes, and I was sad to see it go unmerged. I played this version of cult several times during test-merges and it fixed a lot of the issues. I've put most of Robustin's changes in here.

As far as balance is concerned in retrospect given the small changes in the past few months, I reviewed them all, and I don't believe that the changes in this PR need any modification. Willing to continue tweaking Cult as necessary.

On the Bastard Sword: It's a really weird unintuitive item that a lot of people complain about. Naksu told me to remove cult's ability to make it, so that's what I've done. Cult in general is filled to the brim with bloat items that do something slightly differently without being very valuable so it's not a great loss.

lilacrhododendron committed:
robustin cult nerf except it compiles

Wilchenx commented:
![image](https://user-images.githubusercontent.com/25509881/44127038-9e4c64a4-a03b-11e8-9fe9-175e2fdd0030.png)

oof

subject217 commented:
The only check causing trouble is MapDiffBot. This PR contains zero map edits, so something's fucky with it. It has nothing to do with the PR itself.

ExcessiveUseOfCobblestone commented on code/__DEFINES/cult.dm:
```diff
@@ -10,5 +10,14 @@
+#define MAX_BLOODCHARGE 4
+#define RUNELESS_MAX_BLOODCHARGE 1
+#define CULT_RISEN 0.2
+#define CULT_ASCENDENT 0.4
```
15 and/or 16 need a comment stating what they do

ExcessiveUseOfCobblestone replied:
`Percent before Rise/Ascend` would be sufficient.

ExcessiveUseOfCobblestone commented on code/game/gamemodes/cult/cult.dm:
```diff
@@ -38,8 +38,8 @@
-	required_enemies = 4
-	recommended_enemies = 4
+	required_enemies = 3
+	recommended_enemies = 3
```
undocumented?

vuonojenmustaturska replied:
this was IIRC a fix for a situation where the math worked in a funny way to prevent the gamemode from launching

subject217 replied:
I can remove it or I can add a changelog, won't put it in the CL though unless I understand what exactly it's there for.

ExcessiveUseOfCobblestone commented on code/game/objects/items/storage/book.dm:
```diff
@@ -190,6 +190,20 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+		var/obj/item/soulstone/SS = A
+		to_chat(user, "<span class='notice'>You begin to exorcise [SS].</span>")
+		playsound(src,'sound/hallucinations/veryfar_noise.ogg',40,1)
+		if(do_after(user, 40, target = SS))
```
target should be user (target is for location of progress bar only)

ExcessiveUseOfCobblestone commented on code/game/objects/items/storage/book.dm:
```diff
@@ -190,6 +190,20 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+		if(do_after(user, 40, target = SS))
+			playsound(src,'sound/effects/pray_chaplain.ogg',60,1)
+			SS.usability = TRUE
+			for(var/mob/living/simple_animal/shade/EX in SS)
```
soulstones will only ever have a shade in them outside of foolish mememins, so make it typeless.

Surprised soulstones just don't have a trapped_mob var or something

vuonojenmustaturska replied:
disagree, this is fine

ExcessiveUseOfCobblestone replied:
reeee when would a soulstone ever have anything outside of a shade >:(

ExcessiveUseOfCobblestone commented on code/modules/antagonists/cult/blood_magic.dm:
```diff
@@ -76,15 +82,47 @@
 		if(ishuman(owner))
 			var/mob/living/carbon/human/H = owner
-			H.bleed(40 - rune*32)
+			H.bleed(40 - rune*31)
```
undocumented?

ExcessiveUseOfCobblestone commented on code/modules/antagonists/cult/blood_magic.dm:
```diff
@@ -179,7 +217,7 @@
 	button_icon_state = "transmute"
 	magic_path = "/obj/item/melee/blood_magic/construction"
-	health_cost = 10
+	health_cost = 12
```
undocumented?

subject217 replied:
fixed in changelog

ExcessiveUseOfCobblestone commented on code/modules/antagonists/cult/blood_magic.dm:
```diff
@@ -586,10 +624,13 @@
-			SEND_SOUND(user, sound('sound/effects/magic.ogg',0,1,25))
+			playsound(T, 'sound/machines/airlockforced.ogg', 50, 1)
+			do_sparks(5, TRUE, target)
+			if(do_after(user, 50, target = target))
```
target user

ExcessiveUseOfCobblestone commented on code/modules/antagonists/cult/cult.dm:
```diff
@@ -215,7 +234,52 @@
+		cult_ascendent = TRUE
+		
+		
+/datum/team/cult/proc/rise(cultist)
```
suggestion: 1 proc for both rise/ascend with a second argument (there is not ever a time you will ascend without rising, but you will always rise before ascending)

ExcessiveUseOfCobblestone replied:
you could actually replace the proc with `try_to_rise(mob/living/cultist)` and put the checks in there, so the copypaste of `if(cult_team.cult_risen)` blah blah blah is not necessary.

subject217 replied:
you're welcome to code this but if a maintainer doesn't have a problem with it i'd really rather leave it as is

ExcessiveUseOfCobblestone replied:
only problem with it currently is that there's the same copypasta thrown around so if someone wanted to add a tertiary item, or simply edit the current checks, they'd have to sift through multiple areas which gives avenues to bugs.

I don't mind free GBP tho :^)

ExcessiveUseOfCobblestone commented on code/modules/antagonists/cult/cult_items.dm:
```diff
@@ -960,7 +960,7 @@
 
 /obj/item/shield/mirror/IsReflect()
-	if(prob(block_chance))
+	if(prob(block_chance+20))
```
undocumented?

ExcessiveUseOfCobblestone commented on code/modules/antagonists/cult/runes.dm:
```diff
@@ -229,8 +229,13 @@ structure_check() searches for nearby cultist structures required for the invoca
+/proc/mouse_test(mob/user)
+	var/list/paramlist = params2list(user.client.mouseParams)
+	for(var/i in paramlist)
+		to_chat(world, "Param is [i] with contents [paramlist[i]]")
```
you can just delete this, looks like debug hooplah

that or move it to... click *shudders*

ExcessiveUseOfCobblestone reviewed (changes requested):
Some (albeit minor) undocumented changes :(

I didn't review the tooltip stuff cuz that is ^^ here and I am ___ here

subject217 commented:
looks like robustin was tinkering with shit and left the PR in an unfinished state, that explains why it wasn't merged :^)

i'll push some fixes when i get some spare time

subject217 commented:
In my experience it's only really good if you get a bunch of people trapped in a small area. Otherwise, you can just kite, and the mirror shield with the regular cult sword is better. With the deconversion changes that happened in previous PRs and this PR massively nerfing mass conversion and making several smaller nerfs to various Cult combat mechanics, it's probably not an issue anymore. Assuming I can fix this and get it merged.

HydroFloric commented:
Holy shit finally, maybe once this gets merged nu cult wont be as shit.

Tlaltecuhtli commented:
can you add an examinate desc effect when someone is deconverted that says he got deconverted and it stays for like 2 min? sec can barely catch cultists and usually will holy water 3-4 times the same person because they arent sure he got deconverted or not

ExcessiveUseOfCobblestone commented:
@Tlaltecuhtli with this they shouldn't have the outline/eyes anymore.

BeeSting12 commented:
The two people that downvoted probably dont play otherwise they'd know how stupid powerful cult is.

vuonojenmustaturska commented:
As I said earlier,  feel free to cut the tooltip stuff

lilacrhododendron committed:
removes the tooltip changes

hopefully I won't have to touch other files

lilacrhododendron committed:
removes the ability to make the bastard sword

requested by Naksu

lilacrhododendron committed:
more tooltip shit I missed 

who needs to see if it compiles? that's for losers

subject217 commented:
oh boy

at this point it's probably less work for me to revert and redo the diffs with the dumb fluff cut out and the bastard sword removal

lilacrhododendron committed:
blood_magic.dm patch

wow! I couldn't read one whole diff to catch everything I needed to change!

lilacrhododendron committed:
easier way of doing this 



technically this is bloat but it will compile

lilacrhododendron committed:
missed another

quality coding

lilacrhododendron committed:
fixes empty diff

begone

lilacrhododendron committed:
removes robustin's unlisted reflection buff

what a fraud

lilacrhododendron committed:
deletes debug hooplah

i actually checked to see if this one compiles and yeah it does...

JJRcop commented:
I can squash everything for you if you want

subject217 commented:
I know how to squash commits, this isn't web-edited. At first I was just doing it that way because it was lazy, now I'm doing it because I got burned on squashing commits and had to manually edit diffs out instead of just reverting a commit. So I'm doing it this way.

lilacrhododendron committed:
user shit

hopefully this is alright, it compiles

lilacrhododendron committed:
removes undocumented changes

these are tiny and I can't justify them being in here, so I'm removing them.

subject217 commented:
Tentatively ready. I really can't test a lot of this shit locally, so a test merge would be nice.

lilacrhododendron committed:
appeases the cobblestone

still not convinced this is necessary but it's a trivial change so who cares

AnturK commented on code/modules/antagonists/cult/cult.dm:
```diff
@@ -215,7 +234,52 @@
+				++cultplayers
+			else
+				++alive
+	var/ratio = cultplayers/alive
```
Technically alive can be zero here.

subject217 replied:
This compiles, but doesn't that divide by zero? Or does BYOND just not care if it's 0/0?

AnturK replied:
Not sure what you mean but this is not compile time value, so byond has no idea if this is division by zero.

AnturK replied:
If there's no cultyplayers or alive you can just return since it will won't need update then.

AnturK commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -80,8 +80,11 @@
+	if(!(SLOT_GLASSES in obscured))
+		if(glasses)
+			msg += "[t_He] [t_has] [glasses.get_examine_string(user)] covering [t_his] eyes.\n"
+		else if(eye_color == BLOODCULT_EYE && iscultist(src))
```
This is kinda leaky if we allow pure red as eye color

subject217 replied:
That's what the examine text is for, no? It makes sure there's no doubt that they're a cultist.

ChangelingRain replied:
Pure red is allowed, so people who become cultists with naturally pure red eyes will have this desc.

subject217 replied:
I understand the issue. It seems pretty unlikely to happen unless you were explicitly trying to make it happen. I'm not sure how to fix this. I suppose I could check to see if the cult was already risen, or maybe it's possible to make BLOODCULT_EYE something that isn't selectable as an eye color, but the more issues with this PR the more I feel like I'm in over my head with code that I only partially understand.

AnturK replied:
You can just add simple mob trait for this.

ChangelingRain replied:
Yeeeah you underestimate how many people have pure red eyes(the answer is too many)

ExcessiveUseOfCobblestone replied:
I think my sybil character does actually

AnturK replied:
Up to you, might be nice for admin abuse if it doesn't check specific color.

lilacrhododendron committed:
does this even work

not tested ingame ecksdee please let me checkout

lilacrhododendron committed:
trait shit???

compiled and tested, works as intended
not 100% positive about the trait removals

subject217 commented on code/modules/antagonists/cult/cult.dm:
```diff
@@ -123,7 +127,12 @@
+		H.eye_color = initial(H.eye_color)
+		H.dna.update_ui_block(DNA_EYE_COLOR_BLOCK)
+		H.cut_overlays()
+		H.regenerate_icons()
```
This one I can't figure out. This is for removing the halo and the eyes upon deconversion, right? As far as I can tell the halo overlay isn't tied to the head. I looked around a little bit and I figured out how to cut a specific overlay, but I haven't found any way to just update the head icons. Or maybe I'm misunderstanding.

AnturK replied:
Just needs `update_body` call instead.

subject217 commented:
attempting to bait a review/update on my questions...........

Mey-Ha-Zah commented:
Yeah I'm gonna say, the spell nerfs and the removal of the bastard sword is okay, but the fucking hell sign over every head is bullshit, revs don't get that kinda of "OH TARGET ME". It just serves to make targets for the crew, the red eyes I can stand, maybe force them to only speak Nar Nar after a certain percent, but a HUGE FUCKING Emblem marking them as a target is the worst possible thing. It makes any misdirection impossible.
